Symfony/CS/Fixer/Contrib/EregToPregFixer.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+/*
+ * This file is part of the Symfony CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Matteo Beccati <matteo@beccati.com>
+ */
+class EregToPregFixer extends AbstractFixer
+{
+    /**
+     * @var array the list of the ext/ereg function names, their preg equivalent and the preg modifier(s), if any
+     *            all condensed in an array of arrays.
+     */
+    private static $functions = array(
+        array('ereg', 'preg_match', ''),
+        array('eregi', 'preg_match', 'i'),
+        array('ereg_replace', 'preg_replace', ''),
+        array('eregi_replace', 'preg_replace', 'i'),
+        array('split', 'preg_split', ''),
+        array('spliti', 'preg_split', 'i'),
+    );
+
+    /**
+     * @var array the list of preg delimiters, in order of preference.
+     */
+    private static $delimiters = array('/', '#', '!');
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        if (!$this->cursoryMatch($content)) {
+            return $content;
+        }
+
+        $tokens = Tokens::fromCode($content);
+
+        foreach (self::$functions as $map) {
+            // the sequence is the function name, followed by "(" and a quoted string
+            $seq = array(array(T_STRING, $map[0]), '(', array(T_CONSTANT_ENCAPSED_STRING));
+
+            $currIndex = 0;
+            while (null !== $currIndex) {
+                $ereg = $tokens->findSequence($seq, $currIndex, $tokens->count(), false);
```

Comment:
You are calling `->count()` inside a loop that is inside a loop...
Please cache it ;)